irr/include/vector3d.h

```diff
@@ -33,6 +33,12 @@ class vector3d
 	explicit constexpr vector3d(T n) :
 			X(n), Y(n), Z(n) {}
 
+	template <class U>
+	constexpr static vector3d<T> from(const vector3d<U> &other)
```

Comment:
Amazing!
Can we also have this for vector2d?

Reply:
we do, I actually stole it from there